services/static-webserver/client/source/class/osparc/store/AppSummary.js

```diff
@@ -0,0 +1,35 @@
+/* ************************************************************************
+
+   osparc - the simcore frontend
+
+   https://osparc.io
+
+   Copyright:
+     2024 IT'IS Foundation, https://itis.swiss
+
+   License:
+     MIT: https://opensource.org/licenses/MIT
+
+   Authors:
+     * Odei Maiz (odeimaiz)
+
+************************************************************************ */
+
+qx.Class.define("osparc.store.AppSummary", {
+  type: "static",
+
+  statics: {
+    getLatestUIFromBE: async function() {
+      const params = {
+        url: {
+          productName: osparc.product.Utils.getProductName()
+        }
+      };
+      const appSummary = await osparc.data.Resources.fetch("appSummary", "get", params);
+      if (appSummary && "environment" in appSummary && "osparc.vcsRefClient" in appSummary["environment"]) {
```

Comment:
FYI: I see you do these checks very frequently and I wondered whether javascript had a more compact way to  do the checks. It ssems there is a technique called optional chaining and nullish coalescing operators . It hsould be like
```javascript
return appSummary?.environment?.["osparc.vcsRefClient"] ?? null;
```

Reply:
[Beautiful](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining). I wasn't aware 👌